src/registries/uniswapv2.ts

```diff
@@ -25,6 +25,7 @@ export class RegistryUniswapV2 extends Registry {
 	}
 
 	findPairs = async (tokenWhitelist: Address[]): Promise<Pair[]> =>  {
+		const chainId = await this.web3.eth.getChainId()
```

Comment:
make each registry as part of `findPairs` grab the chainID exactly once, and instantiate all pairs with that chainId